generative_data_prep/utils/arg_parser.py

```diff
@@ -0,0 +1,378 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+
+This module implements argument parsing utilities for a data preparation pipeline.
+
+It provides functions to set up command-line arguments for two main scripts:
+1. data_prep.py: For processing and tokenizing input data.
+2. pipeline.py: For running the entire data preparation pipeline, including splitting data into train/dev/test sets.
+
+The module defines three main functions:
+- data_prep_arg_builder: Adds arguments required for data_prep.py
+- add_data_prep_args: Creates the argument parser for data_prep.py
+- add_pipeline_args: Creates the argument parser for pipeline.py
+- get_arg_parser: Returns the complete argument parser for the entire pipeline
+"""
+import argparse
+from multiprocessing import cpu_count
+
+from .arg_configs import PackingConfig
+from .constants import TOKENIZER_CLASSES, BoundaryType
+
+
+def data_prep_arg_builder(parser: argparse.ArgumentParser):
+    """Adds all the arguments that are required for data_prep.py's argparser, besides the output_path.
+
+    Args:
+        parser (argparse.ArgumentParser): parser to add arguments to
+    """
+    parser.add_argument(
+        "--input_path",
+        type=str,
+        required=False,
+        help="The input jsonl file path or a path to a directory of input jsonls.",
+    )
+    parser.add_argument(
+        "--input_file_path",
+        type=str,
+        required=False,
+        help="Deprecated argument",
+    )
+    parser.add_argument(
+        "--output_path",
+        type=str,
+        required=True,
+        help=(
+            "The path to the output directory if using end to end data preparation, the path to output hdf5 file if"
+            " running tokenization"
+        ),
+    )
+    parser.add_argument(
+        "--log_file_path",
+        type=str,
+        required=False,
+        default=None,
+        help="The log file path. Defaults to '<output_path>/logs.log'",
+    )
+    parser.add_argument(
+        "--overwrite_output_path",
+        action="store_true",
+        help="If the file or files stored at the output path can be over-written",
+    )
+
+    parser.add_argument(
+        "--ignore_input_format_error",
+        action="store_true",
+        help=(
+            "Ignore format errors in the input jsonl file by skipping misformatted lines."
+            " Warning: these lines are dropped from the generated dataset"
+        ),
+    )
+
+    parser.add_argument(
+        "--tokenizer_class",
+        type=str,
+        choices=list(TOKENIZER_CLASSES.keys()),
```

Comment:
Can you elaborate why we need to default this to GPT2. Shouldn't this be a required argument ?